src/main/java/ru/timeconqueror/tcneiadditions/util/TCUtil.java

```diff
@@ -127,4 +134,100 @@ public static List<ItemStack> getOreDictionaryMatchingItemsForInfusion(ItemStack
+            for (String researchKey : researchItem.parentsHidden) {
+                String researchName =
+                        ResearchCategories.getCategoryName(ResearchCategories.getResearch(researchKey).category)
+                                + " : "
+                                + ResearchCategories.getResearch(researchKey).getName();
+                if (ResearchManager.isResearchComplete(playerName, researchKey)) {
+                    if (researchItem.parentsHidden.length <= 10) {
+                        researchName = EnumChatFormatting.GREEN + "" + EnumChatFormatting.STRIKETHROUGH + researchName;
+                        list.add(EnumChatFormatting.RESET + "    " + researchName);
+                    }
+                } else {
+                    needResearch++;
+                    researchName = EnumChatFormatting.RED + researchName;
+                    list.add(EnumChatFormatting.RESET + "    " + researchName);
+                }
+            }
+            if (researchItem.parentsHidden.length > 10 && needResearch == 0) {
+                list.add(EnumChatFormatting.GREEN + "    "
+                        + StatCollector.translateToLocal("tcneiadditions.research.prerequisites.allresearched"));
+            }
+        }
+        // Item scan
+        if (researchItem.getItemTriggers() != null && researchItem.getItemTriggers().length != 0) {
+            list.add(StatCollector.translateToLocal("tcneiadditions.research.prerequisites.item") + ":");
+            for (ItemStack itemStack : researchItem.getItemTriggers()) {
+                String displayName = itemStack.getDisplayName();
+                list.add("    " + displayName);
+            }
+        }
+        // Entity scan
+        if (researchItem.getEntityTriggers() != null && researchItem.getEntityTriggers().length != 0) {
+            list.add(StatCollector.translateToLocal("tcneiadditions.research.prerequisites.entity") + ":");
+            for (String entityKey : researchItem.getEntityTriggers()) {
+                String entityName = StatCollector.translateToLocal("entity." + entityKey + ".name");
+                list.add("    " + entityName);
+            }
+        }
+        // Aspect scan
+        if (researchItem.getAspectTriggers() != null && researchItem.getAspectTriggers().length != 0) {
+            list.add(StatCollector.translateToLocal("tcneiadditions.research.prerequisites.aspect") + ":");
+            for (Aspect aspect : researchItem.getAspectTriggers()) {
+                String aspectName = aspect.getName() + " - " + aspect.getLocalizedDescription();
+                list.add("    " + aspectName);
+            }
+        }
+        // Kill scan
+        if (Loader.isModLoaded("Automagy") && researchItem.category.equals("AUTOMAGY")) {
+            Set<String> killList = getKeysByValue(ModResearchItems.cluesOnKill, researchItem.key);
```

Comment:
As far as I know this will still crash if Automagy is not installed. Either reference this class indirectly (either via reflection (meh) or by making a "compat class" which handles the direct interaction with Automagy and is only called from here if Automagy is loaded (good)) or by making Automagy a hard-dep (bad!).

Also please cache whether Automagy is loaded or not and add it as soft-dep in the `@Mod`-annotation.

Reply:
1. i haven't seen a crash even if Automagy is not loaded.
2. i don't understand the compat class, can you give an example?
3. did you mean after:Automagy?

Reply:
This works just fine afaik. However, storing result of `Loader#isModLoaded` is a good thing. It's essentially `Map#get` so super fast query, but we can reuse the value somewhere else in the future.